sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/checkpoint/EventCheckpointManager.java

```diff
@@ -4,62 +4,25 @@
 package com.azure.spring.eventhubs.checkpoint;
 
 import com.azure.messaging.eventhubs.EventData;
+import com.azure.messaging.eventhubs.models.EventBatchContext;
 import com.azure.messaging.eventhubs.models.EventContext;
 import com.azure.spring.eventhubs.support.EventDataHelper;
 import com.azure.spring.messaging.checkpoint.CheckpointConfig;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * @author Warren Zhu
- * @author Xiaolu Dai
+ *
  */
-public abstract class CheckpointManager {
-    private static final Logger LOG = LoggerFactory.getLogger(CheckpointManager.class);
+public abstract class EventCheckpointManager implements EventCheckpoint, EventBatchCheckpoint {
```

Comment:
It's weird to give EventCheckpointManager the ability of EventBatchCheckpoint

Reply:
But it's acceptable.

Reply:
yes, it is weird. ..need to redesign interface for this, for now too much hard code changes for legacy code for refactoring